### PR TITLE
Rework GPU pipeline.

### PR DESCRIPTION
**Pull request by jbruestle**

### CODE REVIEW

**Comment by jbruestle** [comment]
Hmm, meant to open this as WIP, but misclicked, not sure how to change it.

**Comment by jbruestle** [comment]
Remaining major TODOs:
1) Consider breaking into multiple PRs (esp some of the clear utility goo like autotile)
2) Rename map_affine_parallel to something else (talk with @flaub) and relocate.

File: pmlc/dialect/pxa/transforms/passes.td [hunk 1]
```diff
@@ -6,6 +6,11 @@ include "mlir/Pass/PassBase.td"
 def AffineNormalize : FunctionPass<"pxa-normalize"> {
   let summary = "Normalize affine ops";
   let constructor = "pmlc::dialect::pxa::createAffineNormalizePass()";
+  let options = [
+    Option<"promote", "promote-threshold", "bool",
```
Comment (flaub) [inline pmlc/dialect/pxa/transforms/passes.td:10]: Suggest rename to just `promote`

File: pmlc/target/intel_gen/pipeline.cc [hunk 2]
```diff
@@ -33,37 +32,61 @@ namespace {
...
 
+  // Lower to PXA
   pm.addPass(conversion::tile_to_pxa::createLowerTileToPXAPass());
+  pm.addPass(pmlc::dialect::pxa::createAffineNormalizePass());
   pm.addPass(createCanonicalizerPass());
   pm.addPass(createCSEPass());
+
+  // Move accumulation indexes into an inner loop
   pm.addPass(pmlc::dialect::pxa::createTileAccumulatePass());
+  pm.addPass(pmlc::dialect::pxa::createAffineNormalizePass(false));
```
Comment (flaub) [inline pmlc/target/intel_gen/pipeline.cc:49]: use `/*promote=*/false` to make it more clear what that parameter is all about.

File: pmlc/target/intel_gen/pipeline.cc [hunk 3]
```diff
@@ -33,37 +32,61 @@ namespace {
...
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createCSEPass());
+
+  // Assign GPU blocks + threads to outermost loop
+  pm.addPass(pmlc::dialect::pxa::createGPUThreadPass(256));
```
Comment (flaub) [inline pmlc/target/intel_gen/pipeline.cc:54]: ditto here

**Pull request merged**